pm/recipient_test.go

```diff
@@ -0,0 +1,436 @@
+package pm
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func TestReceiveTicket(t *testing.T) {
+	sender := ethcommon.HexToAddress("A69cdA26600c155cF2c150964Bdb5371ac3f606F")
+	b := newStubBroker()
+	v := &stubValidator{}
+	ts := newStubTicketStore()
+	secret := [32]byte{3}
+	faceValue := big.NewInt(100)
+	winProb := big.NewInt(100)
+	sig := []byte("foo")
+
+	r := NewRecipient(b, v, ts, secret, faceValue, winProb)
+
+	// Config stub validator with valid non-winning tickets
+	v.SetIsValidTicket(true)
+
+	params, err := r.TicketParams(sender)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Test invalid recipientRand from seed (invalid seed)
+	ticket := &Ticket{
+		Recipient:         ethcommon.Address{},
+		Sender:            sender,
+		FaceValue:         faceValue,
+		WinProb:           winProb,
+		SenderNonce:       0,
+		RecipientRandHash: params.RecipientRandHash,
+	}
+
+	// Using invalid seed
+	_, err = r.ReceiveTicket(ticket, sig, big.NewInt(12312))
+	if err == nil {
+		t.Error("expected invalid recipientRand from seed error")
+	}
+	if err != nil && err != errInvalidRecipientRandFromSeed {
+		t.Errorf("expected invalid recipientRand from seed error, got %v", err)
```

Comment:
I was thinking that it might be more helpful for real-life debugging to include variable values in the error strings we generate. In this case, I would suggest we format the error return value as: `invalid recipientRand generated from seed. Sender: <senderAddress>, Seed: <seed value>`, perhaps with a few more data points.

Wdyt of this approach versus always returning the same error string?

Reply:
Sure, we can format the error string. The downside with formatting the error string is that we will no longer be able to do simple constant error checks while testing and instead we'll have to type out error strings or do string matching.

Reply:
We're having the same conversation on both our PRs :) Let's resolve this dilemma in one place and apply our conclusions in both places. Where would you like our main thread about const error strings vs dynamic error string params to take place?